web/packages/teleport/src/Bots/Add/GitHubActionsK8s/KubernetesLabelsSelect.tsx

```diff
@@ -0,0 +1,571 @@
+        message:
+          'At least one label must be selected. Use wildcard *:* to match any label.',
+      };
+    }
+    return {
+      valid: true,
+    };
+  });
+
+  const handleChanged = (labels: KubernetesLabel[]) => {
+    setShowPicker(false);
+    onChange(labels);
+  };
+
+  const handleCancel = () => {
+    setShowPicker(false);
+  };
+
+  return (
+    <Box {...styles}>
+      <LabelInput mb={0} aria-label="Labels">
+        <LabelContent>Labels</LabelContent>
+
+        <LabelsContainer aria-describedby={'labels-select-helper-text'}>
+          {selected.map(l => (
+            <Label key={l.name} label={formatLabel(l)} style="secondary" />
+          ))}
+
+          {selected.length === 0 && <EmptyText>No labels selected.</EmptyText>}
+
+          <ButtonIcon
+            onClick={() => {
+              setShowPicker(true);
+            }}
+            aria-label="edit"
+          >
+            <Pencil size="medium" />
+          </ButtonIcon>
+        </LabelsContainer>
+      </LabelInput>
+
+      <HelperTextLine
+        hasError={!valid}
+        helperTextId={'labels-select-helper-text'}
+        errorMessage={message}
+      />
+
+      {showPicker && (
+        <Picker
+          selected={selected}
+          onChange={handleChanged}
+          onCancel={handleCancel}
+        />
+      )}
+    </Box>
+  );
+}
+
+const LabelsContainer = styled(Flex)`
+  flex-wrap: wrap;
+  overflow: hidden;
+  gap: ${props => props.theme.space[2]}px;
+  align-items: center;
+`;
+
+const LabelText = styled(Text).attrs({
+  typography: 'body2',
+})`
+  white-space: nowrap;
+`;
+
+function Picker(props: {
+  selected: KubernetesLabel[];
+  onChange: (labels: KubernetesLabel[]) => void;
+  onCancel: () => void;
+}) {
+  const { selected: initial, onChange, onCancel } = props;
+
+  const [selected, setSelected] = useState(initial);
+  const [manualInput, setManualInput] = useState('');
+
+  const ctx = useTeleport();
+  const flags = ctx.getFeatureFlags();
+  const hasListPermission = flags.kubernetes;
+
+  const {
+    data,
+    isLoading,
+    isFetchingNextPage,
+    error,
+    hasNextPage,
+    fetchNextPage,
+  } = useInfiniteQuery({
+    enabled: hasListPermission,
+    queryKey: ['list', 'unified_resources', cfg.proxyCluster, ['kube_cluster']],
+    queryFn: ({ pageParam, signal }) =>
+      ctx.resourceService.fetchUnifiedResources(
+        cfg.proxyCluster,
+        {
+          kinds: ['kube_cluster'],
+          startKey: pageParam,
+          limit: 20,
+        },
+        signal
+      ),
+    initialPageParam: '',
+    getNextPageParam: data => data?.startKey || undefined,
+    placeholderData: keepPreviousData,
+    staleTime: 30_000, // Cached pages are valid for 30 seconds
+  });
+
+  const handleAdd = (name: string, value: string) => {
+    setSelected(prev => {
+      const index = prev.findIndex(l => l.name === name);
+      if (index == -1) {
+        return [...prev, { name, values: [value] }];
+      } else {
```

Comment:
The else is not needed

Reply:
I meant as you're returning in the if statement, the else statement isn't needed, and the stuff in the else block can be moved left